lib/isolator/adapters/config.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require "forwardable"
+
+module Isolator
+  module Adapters
+    # Wraps the adapter config and provides accessor by name
+    class Config
+      include Enumerable
+      extend Forwardable
+      def_delegators :hash_config, :each, :[]
+
+      def register(id, adapter)
+        raise "Adapter already registered: #{id}" if respond_to?(id)
+        define_singleton_method(id) { self[id] }
```

Comment:
Let's use the  previous `method_missing` approach but without fallback to `super` (in order to raise `NoMethodError` when no such key).

The provided approach doesn't play well with deleting keys: after a key has been deleted we except to got an exception; and now it returns `nil` instead.